mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+import math
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+from matplotlib.colors import BoundaryNorm
+from matplotlib.ticker import MaxNLocator
+
+from mcstasscript.data.data import McStasMetaData
+from mcstasscript.data.data import McStasPlotOptions
+from mcstasscript.data.data import McStasData
+
+
+class make_plot:
+    """
+    make_plot plots contents of McStasData objects
+
+    Plotting is controlled through options assosciated with the
+    McStasData objects.
+
+    If a list is given, the plots appear individually.
+    """
+
+    def __init__(self, data_list, **kwargs):
+        """
+        plots McStasData, single object or list of McStasData
+
+        The options concerning plotting are stored with the data
+
+        Parameters
+        ----------
+        data_list : McStasData or list of McStasData
+            McStasData to be plotted
+        """
+
+        # Relevant options:
+        #  select colormap
+        #  show / hide colorbar
+        #  custom title / label
+        #  color of 1d plot
+        #  overlay several 1d
+        #  log scale (orders of magnitude)
+        #  compare several 1d
+        #  compare 2D
+
+        if isinstance(data_list, McStasData):
+            # Only a single element, put it in a list for easier syntax later
+            data_list = [data_list]
+
+        number_of_plots = len(data_list)
+
+        if "fontsize" in kwargs:
+            plt.rcParams.update({'font.size': kwargs["fontsize"]})
+
+        print("number of elements in data list = " + str(len(data_list)))
+
+        index = -1
+        for data in data_list:
+            index = index + 1
+
+            print("Plotting data with name " + data.metadata.component_name)
+            if type(data.metadata.dimension) == int:
+                fig = plt.figure(0)
+
+                x_axis_mult = data.plot_options.x_limit_multiplier
+                # print(data.T)
+                x = data.xaxis*x_axis_mult
+                y = data.Intensity
+                y_err = data.Error
+
+                #(fig, ax0) = plt.errorbar(x, y, yerr=y_err)
+                plt.errorbar(x, y, yerr=y_err)
```

Comment:
Then call the methods on the axes, to also be sure they are plotting on the correct axes
```suggestion
                ax0.errorbar(x, y, yerr=y_err)
```

Reply:
All plotting is now done through the ax object instead of plt directly. Great suggestion.